packages-ts/starknet/src/network.ts

```diff
@@ -0,0 +1,122 @@
+import { ChildProcess, spawn } from 'child_process'
+import axios from 'axios'
+
+export abstract class IntegratedDevnet {
+  protected childProcess: ChildProcess
+
+  constructor(protected port: string) {
+    IntegratedDevnet.cleanupFns.push(this.cleanup.bind(this))
+  }
+
+  protected static cleanupFns: Array<() => void> = []
+
+  public static cleanAll(): void {
+    this.cleanupFns.forEach((fn) => fn())
+  }
+
+  protected abstract spawnChildProcess(): Promise<ChildProcess>
+
+  protected abstract cleanup(): void
+
+  private async isServerAlive() {
+    try {
+      await axios.get(`http://127.0.0.1:5050/is_alive`)
+      return true
+    } catch (err: unknown) {
+      // cannot connect, so address is not occupied
+      return false
+    }
+  }
+
+  public async start(): Promise<void> {
+    if (await this.isServerAlive()) {
+      this.cleanup()
+    }
+    this.childProcess = await this.spawnChildProcess()
+
+    // capture the most recent message from stderr
+    // Needed to avoid TimeOut error in some of our tests.
+    this.childProcess.stderr?.on('data', async () => {})
+    return new Promise((resolve, reject) => {
+      setTimeout(resolve, 4000)
+
+      this.childProcess.on('error', (error) => {
+        reject(error)
+      })
+    })
+  }
+
+  public stop() {
+    if (!this.childProcess) {
+      return
+    }
+
+    this.cleanup()
+  }
+}
+
+class VenvDevnet extends IntegratedDevnet {
```

Comment:
There is no need for these `Venv` prefixes, as we don't interact with venvs in any way here. 
Using it to set up our env is just an implementation detail these classes should not know about.

```suggestion
class StarknetDevnet extends ChildProcessManager {
```